### PR TITLE
Feat/input verification

### PR DESCRIPTION
**Pull request by Gurr1**

### CODE REVIEW

**Comment by Portals** [comment]
How do errors look if you don't send correct data for example?

File: backend/src/main/java/it/chalmers/gamma/controller/WhitelistController.java [hunk 1]
```diff
@@ -44,19 +50,17 @@ public WhitelistController(
...
+        }
         if (this.whitelistService.isCIDWhiteListed(cid.getCid())) {
             Whitelist whitelist = this.whitelistService.getWhitelist(cid.getCid());
             String code = this.activationCodeService.generateActivationCode();
             ActivationCode activationCode = this.activationCodeService.saveActivationCode(whitelist, code);
             sendEmail(activationCode);
         }
-
-        /*
-        We always want to send the same response, for security reasons.
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/controller/WhitelistController.java]: Why not keep this comment? I think it's useful

File: backend/src/main/java/it/chalmers/gamma/requests/CreateGroupRequest.java [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
 import java.util.List;
 import java.util.Objects;
 
+import javax.validation.constraints.Email;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
 public class CreateGroupRequest {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/CreateGroupRequest.java]: Add spaces between fields

File: backend/src/main/java/it/chalmers/gamma/requests/CreateITUserRequest.java [hunk 3]
```diff
@@ -4,15 +4,30 @@
...
 
-public class CreateITUserRequest {
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
 
+public class CreateITUserRequest {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/CreateITUserRequest.java]: Spaces between fields

File: backend/src/main/java/it/chalmers/gamma/requests/CreateSuperGroupRequest.java [hunk 4]
```diff
@@ -10,12 +10,12 @@
 
 
 public class CreateSuperGroupRequest {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/CreateSuperGroupRequest.java]: Spaces between fields

File: backend/src/main/java/it/chalmers/gamma/requests/CreateWebsiteRequest.java [hunk 5]
```diff
@@ -1,10 +1,12 @@
 package it.chalmers.gamma.requests;
-
 import java.util.Objects;
+import javax.validation.constraints.NotNull;
 
 public class CreateWebsiteRequest {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/CreateWebsiteRequest.java]: Spaces between fields

File: backend/src/main/java/it/chalmers/gamma/requests/DeleteGroupRequest.java [hunk 6]
```diff
@@ -1,11 +1,14 @@
 package it.chalmers.gamma.requests;
 
 import java.util.Objects;
+import javax.validation.constraints.NotNull;
 
-public class DeleteGroupRequest {
-    String group;
-    String adminUser;
-    String adminPassword;
+public class DeleteGroupRequest {       // Unsure about this one
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/DeleteGroupRequest.java]: Spaces between fields

File: backend/src/main/java/it/chalmers/gamma/requests/CidPasswordRequest.java [hunk 7]
```diff
@@ -1,10 +1,13 @@
 package it.chalmers.gamma.requests;
 
 import java.util.Objects;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
 
 public class CidPasswordRequest {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/CidPasswordRequest.java]: Spaces between fields

File: backend/src/main/java/it/chalmers/gamma/requests/AuthorizationRequest.java [hunk 8]
```diff
@@ -1,10 +1,14 @@
 package it.chalmers.gamma.requests;
 
 import java.util.Objects;
+import javax.validation.constraints.NotNull;
 
 public class AuthorizationRequest {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/AuthorizationRequest.java]: Spaces between fields

File: backend/src/main/java/it/chalmers/gamma/requests/AdminViewCreateITUserRequest.java [hunk 9]
```diff
@@ -1,15 +1,30 @@
 package it.chalmers.gamma.requests;
 
 import java.util.Objects;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Size;
 
 public class AdminViewCreateITUserRequest {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/AdminViewCreateITUserRequest.java]: Spaces between fields

File: backend/src/main/java/it/chalmers/gamma/requests/AddUserGroupRequest.java [hunk 10]
```diff
@@ -1,21 +1,14 @@
 package it.chalmers.gamma.requests;
 
-import java.time.Year;
 import java.util.Objects;
+import javax.validation.constraints.NotEmpty;
 
 public class AddUserGroupRequest {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/AddUserGroupRequest.java]: Spaces between fields

**Comment by NeonSky** [comment]
Tbh complaining about `Spaces between fields` is pretty meaningless. @Portals if you want this to be a thing, then it should be added to CI and resolved in another branch. It is totally unrelated to fixing input verification.

**Comment by Portals** [comment]
@NeonSky 
I don't think it's unrelated. Usually, it looks great to have code like this:

```java
private float x;
private String s;
```

But when there's a lot of annotations, it can feel quite cluttered. The need to always separate fields by one row could actually make the reading of code worse. Since you cannot easily group together fields that are relevant to each other.

**Pull request merged**